apm-sniffer/optional-plugins/shenyu-2.4.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/shenyu/v24x/define/AlibabaDubboPluginInstrumentation.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.shenyu.v24x.define;
+
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.ConstructorInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.InstanceMethodsInterceptPoint;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.ClassInstanceMethodsEnhancePluginDefine;
+import org.apache.skywalking.apm.agent.core.plugin.match.ClassMatch;
+
+import static net.bytebuddy.matcher.ElementMatchers.named;
+import static org.apache.skywalking.apm.agent.core.plugin.match.NameMatch.byName;
+
+public class AlibabaDubboPluginInstrumentation extends ClassInstanceMethodsEnhancePluginDefine {
+    @Override
+    protected ClassMatch enhanceClass() {
+        return byName("org.apache.shenyu.plugin.alibaba.dubbo.AlibabaDubboPlugin");
```

Comment:
Why does Shenyu require old Dubbo? As a proxy, this should be transparent, isn't it?

Reply:
I have tested that Dubbo 2.7x client can access the old version of Dubbo, but Shenyu is based on two plug-ins, not one.
I'll give a suggestion to Shenyu community.

Reply:
Alibaba Dubbo is like 3 years old, I guess? And considering no one is maintaining it, and fewer people are going to use it, I would suggest we remove that support.
Even for SkyWalking's native Alibaba Dubbo plugin, once we received a report saying there is a bug, the official solution would be to remove it too.

Reply:
> Alibaba Dubbo is like 3 years old, I guess? And considering no one is maintaining it, and fewer people are going to use it, I would suggest we remove that support. Even for SkyWalking's native Alibaba Dubbo plugin, once we received a report saying there is a bug, the official solution would be to remove it too.

Agree. In addition, the new version of Dubbo client can also call the old duboo service, although I haven't fully tested it

I will remove support for the old version.

Reply:
> Agree. In addition, the new version of Dubbo client can also call the old duboo service, although I haven't fully tested it

I think the protocol is not unchanged, RPC frameworks are keeping forward compatibility seriously, I assume Dubbo community should consider this already. 
If you or Shenyu community has concerns, you could submit a question to Dubbo to confirm, I think they would be happy to confirm and see 3 projects closing integration.